ext/zlib/zlib.c

```diff
@@ -555,7 +559,233 @@ rb_zlib_crc_table(VALUE obj)
     return dst;
 }
 
+static VALUE
+crc32c(unsigned long crc, const unsigned char *buf, size_t size)
+{
+    const uint8_t *p = buf;
+
+    crc = crc ^ 0xFFFFFFFF;
+    while (size--)
+        crc = crc32ctbl[(crc ^ *p++) & 0xff] ^ (crc >> 8);
+	return crc ^ 0xFFFFFFFF;
+}
+
+/*
+ * Document-method: Zlib.crc32c
+ *
+ * call-seq: Zlib.crc32c(string, crc)
+ *
+ * Calculates CRC32C checksum for +string+, and returns updated value of +crc+. If
+ * +string+ is omitted, it returns the CRC32C initial value. If +crc+ is omitted, it
+ * assumes that the initial value is given to +crc+. If +string+ is an IO instance,
+ * reads from the IO until the IO returns nil and returns CRC32C checksum of all read
+ * data.
+ *
+ * FIXME: expression.
+ */
+static VALUE
+rb_zlib_crc32c(int argc, VALUE *argv, VALUE klass)
+{
+    return do_checksum(argc, argv, &crc32c);
+}
+
+/*
+ * The following CRC32C code was adapted from the CRC32 code in zlib, including
+ * the surrounding functions.
+ * https://github.com/luvit/zlib/blob/8de57bce969eb9dafc1f1f5c256ac608d0a73ec4/crc32.c#L355
```

Comment:
https://zlib.net/ states https://github.com/madler/zlib is their official repository?

Reply:
Good catch, can't always trust google.. I'll update this comment and the code where applicable.